From aef763366b91a2ca8122c546d688b4c0e99c489e Mon Sep 17 00:00:00 2001
From: Thejas Nair <thejas@hortonworks.com>
Date: Mon, 27 Apr 2015 13:38:32 -0700
Subject: [PATCH 0269/1164] HIVE-10465 : whitelist restrictions don't get
 initialized in new copy of HiveConf (Thejas Nair,
 reviewed by Daniel Dai)

---
 .../java/org/apache/hadoop/hive/conf/HiveConf.java |    1 +
 .../TestJdbcWithSQLAuthorization.java              |   25 ++++++++++++++++++++
 2 files changed, 26 insertions(+)

diff --git a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
index d4ce9e9..d54c5da 100644
--- a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
+++ b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
@@ -2566,6 +2566,7 @@ public HiveConf(HiveConf other) {
     isSparkConfigUpdated = other.isSparkConfigUpdated;
     origProp = (Properties)other.origProp.clone();
     restrictList.addAll(other.restrictList);
+    modWhiteListPattern = other.modWhiteListPattern;
   }
 
   public Properties getAllProperties() {
diff --git a/itests/hive-unit/src/test/java/org/apache/hive/jdbc/authorization/TestJdbcWithSQLAuthorization.java b/itests/hive-unit/src/test/java/org/apache/hive/jdbc/authorization/TestJdbcWithSQLAuthorization.java
index 6f1a8b2..bd0ba0e 100644
--- a/itests/hive-unit/src/test/java/org/apache/hive/jdbc/authorization/TestJdbcWithSQLAuthorization.java
+++ b/itests/hive-unit/src/test/java/org/apache/hive/jdbc/authorization/TestJdbcWithSQLAuthorization.java
@@ -175,4 +175,29 @@ private void checkAssertContains(String expectedSubString, String message) {
     fail("Message [" + message + "] does not contain substring [" + expectedSubString + "]");
   }
 
+  @Test
+  public void testConfigWhiteList() throws Exception {
+
+    // create tables as user1
+    Connection hs2Conn = getConnection("user1");
+
+    Statement stmt = hs2Conn.createStatement();
+    try {
+      stmt.execute("set hive.metastore.uris=x");
+      fail("exception expected");
+    } catch (SQLException e) {
+      String msg = "Cannot modify hive.metastore.uris at runtime. "
+          + "It is not in list of params that are allowed to be modified at runtime";
+      assertTrue(e.getMessage().contains(msg));
+    }
+
+    stmt.execute("set hive.exec.reducers.bytes.per.reducer=10000");
+    //no exception should be thrown
+
+    stmt.close();
+    hs2Conn.close();
+  }
+
+
+
 }
-- 
1.7.9.5

